### ISSUE TITLE
`Authorization` header is stripped from queries.

### ISSUE THREAD

**Issue by J61n** [drwpow/openapi-typescript#1225]
**Description**

When I try to perform a query, it seems like the `Authorization` header is not sent with the query. This bug does not seem to occur when setting a header with a different name.

**Reproduction**

I tried setting the header as follows:

Option 1, on the client (similar to the [documentation](https://openapi-ts.pages.dev/openapi-fetch/examples/#authentication):
```ts
const { get } = createClient<paths>({
  headers: {
    Authorization: `Bearer ${someToken}`,
  },
});
const { data } = await get("/api/something", {
  params: {},
});
```

Option 2, on the query (as I'm used to from using the fetch API):
```ts
const { get } = createClient<paths>();
const { data } = await get("/api/something", {
  params: {},
  headers: {
    Authorization: `Bearer ${someToken}`,
  },
});
```

**Actual result**

The `GET` request is sent *without* the `Authorization` header.

**Expected result**

The `GET` request is sent with an `Authorization` header that has a value like `Bearer {...}`.

**Checklist**

- [ ] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-fetch/blob/main/CONTRIBUTING.md))

**Comment by J61n** [comment]
After further research, I found out that this behavior seems to be Firefox-specific. In my case, the frontend development server was configured to proxy requests starting with `/api` to a backend server (HTTP), which is configured to redirect HTTP requests to HTTPS automatically. As mentioned [here](https://github.com/axios/axios/issues/5597), Firefox strips the `Authorization` header by default configuration when a redirect occurs, thus leading to the behavior described above.

**TL;DR:** Ensuring that requests weren't redirected solved the problem.

**Comment by drwpow** [comment]
Ah glad you figured it out.
